src/test/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientTest.java

```diff
@@ -439,16 +495,56 @@ public void onBlock_return_when_node_is_syncing() {
         );
         btcReleaseClient.start(federation);
 
+        List<TransactionReceipt> receipts = new ArrayList<>();
+        TransactionReceipt transactionReceipt = mock(TransactionReceipt.class);
+        receipts.add(transactionReceipt);
+
+        // Act
+        ethereumListener.get().onBlock(null, receipts);
+
+        // Assert
+        verifyZeroInteractions(transactionReceipt);
+    }
+
+    @Test
+    public void onBlock_return_when_pegout_is_disabled() {
+        // Arrange
+        Federation federation = TestUtils.createFederation(params, 1);
+
+        Ethereum ethereum = mock(Ethereum.class);
+        AtomicReference<EthereumListener> ethereumListener = new AtomicReference<>();
+
+        doAnswer((InvocationOnMock invocation) -> {
+            ethereumListener.set((EthereumListener) invocation.getArguments()[0]);
+            return null;
+        }).when(ethereum).addListener(ArgumentMatchers.any(EthereumListener.class));
+
+        FederatorSupport federatorSupport = mock(FederatorSupport.class);
+
+        FedNodeSystemProperties fedNodeSystemProperties = mock(FedNodeSystemProperties.class);
+        doReturn(Constants.regtest()).when(fedNodeSystemProperties).getNetworkConstants();
+        doReturn(false).when(fedNodeSystemProperties).isPegoutEnabled();
+
+        NodeBlockProcessor nodeBlockProcessor = mock(NodeBlockProcessor.class);
+        when(nodeBlockProcessor.hasBetterBlockToSync()).thenReturn(false);
+
+        BtcReleaseClient btcReleaseClient = new BtcReleaseClient(
+            ethereum,
+            federatorSupport,
+            fedNodeSystemProperties,
+            nodeBlockProcessor
+        );
+        btcReleaseClient.start(federation);
 
         List<TransactionReceipt> receipts = new ArrayList<>();
-        TransactionReceipt mocktTransactionReceipt = mock(TransactionReceipt.class);
-        receipts.add(mocktTransactionReceipt);
+        TransactionReceipt transactionReceipt = mock(TransactionReceipt.class);
+        receipts.add(transactionReceipt);
 
         // Act
         ethereumListener.get().onBlock(null, receipts);
 
         // Assert
-        Mockito.verifyZeroInteractions(mocktTransactionReceipt);
+        verifyZeroInteractions(transactionReceipt);
```

Comment:
And here,

Let's add also one more assertion.

```suggestion
        verify(nodeBlockProcessor, never()).hasBetterBlockToSync();
        verifyZeroInteractions(transactionReceipt);
```